tests/test_rp_vars.py

```diff
@@ -53,4 +53,23 @@ def test_get_all_rp_vars(new_solver_session_no_transcript) -> None:
     # case reader comparison
     case = CaseReader(case_filepath=case_path)
     case_vars = case.rp_vars()
-    assert len(case_vars) == pytest.approx(9000, 450)
\ No newline at end of file
+    assert len(case_vars) == pytest.approx(9000, 450)
+
+
+@pytest.mark.dev
+@pytest.mark.fluent_232
+def test_rp_vars_allowed_values(new_solver_session_no_transcript) -> None:
+    solver = new_solver_session_no_transcript
+    rp_vars = solver.rp_vars
+
+    assert rp_vars("number-of-iterations") == 0
+
+    with pytest.raises(RuntimeError) as msg:
+        rp_vars("number-of-iterat")
+
+    assert msg.value.args[0] == "number-of-iterat is not an allowed  name.\n" \
+                                "The most similar names are: number-of-iterations, " \
+                                "number-of-time-steps, lb/number-of-timesteps, " \
+                                "number-of-samples, gpuapp/total-number-of-subiterations."
+
+    assert "number-of-iterations" in rp_vars.allowed_values()
```

Comment:
more precisely it should be rp_vars.var.allowed_values()

Reply:
@seanpearsonuk, could we have a short discussion on this?

Reply:
@prmukherj only just seeing your response here. `allowed_values()` is customarily called on the argument, which is `var` here.